confidential-transfer/proof-generation/src/transfer_with_fee.rs

```diff
@@ -1,3 +1,51 @@
+//! Generates the zero-knowledge proofs required for a confidential transfer with a fee.
+//!
+//! A confidential transfer with a fee is more complex than a simple transfer. It requires five
+//! distinct zero-knowledge proofs to ensure the validity of the transfer, the solvency of the
+//! sender, and the correctness of the fee amount according to the on-chain mint configuration.
+//!
+//! ## Protocol Flow and Proof Components
+//!
+//! 1.  **Fee Calculation**: The client first calculates the required fee based on the transfer
+//!     amount and the on-chain fee parameters (rate and maximum cap).
+//!
+//! 2.  **Encrypt Amounts**: The gross transfer amount and the fee amount are each split into low
+//!     and high bit components. These components are then encrypted into separate grouped (twisted)
+//!     ElGamal ciphertexts with the appropriate decryption handles for the involved parties (source,
+//!     destination, auditor, and withdraw authority).
+//!
+//! 3.  **Generate Proofs**: The sender generates five proofs that work in concert:
+//!
+//!     -   **Transfer Amount Ciphertext Validity Proof
+//!         (`BatchedGroupedCiphertext3HandlesValidityProofData`)**: Certifies that the grouped
+//!         ElGamal ciphertext for the gross transfer amount is well-formed.
+//!
+//!     -   **Fee Ciphertext Validity Proof
+//!         (`BatchedGroupedCiphertext2HandlesValidityProofData`)**: Certifies that the grouped
```

Comment:
Just to make sure, this is only 2 handles because we don't encrypt the fee amount with the auditor key, correct?